node-js-basics-101/node_modules/asynquence-contrib/contrib.src.js

```diff
@@ -0,0 +1,2360 @@
+					;
+				}
+
+				// all segments complete?
+				if (!finished &&
+					completed === fns.length
+				) {
+					finished = true;
+
+					complete(trigger);
+				}
+			}
+
+			var completed = 0, error_messages = [], finished = false,
+				success_messages = [],
+				sq = ASQ.apply(ø,ARRAY_SLICE.call(arguments,1))
+			;
+
+			wrapGate(sq,fns,success,failure,reset);
+
+			sq.pipe(done);
+		});
+
+		return api;
+	};
+});
+// "errfcb"
+ASQ.extend("errfcb",function $$extend(api,internals){
+	return function $$errfcb() {
+		// create a fake sequence to extract the callbacks
+		var sq = {
+			val: function $$then(cb){ sq.val_cb = cb; return sq; },
+			or: function $$or(cb){ sq.or_cb = cb; return sq; }
+		};
+
+		// trick `seq(..)`s checks for a sequence
+		sq[brand] = true;
+
+		// immediately register our fake sequence on the
+		// main sequence
+		api.seq(sq);
+
+		// provide the "error-first" callback
+		return function $$errorfirst$callback(err) {
+			if (err) {
+				sq.or_cb(err);
+			}
+			else {
+				sq.val_cb.apply(ø,ARRAY_SLICE.call(arguments,1));
+			}
+		};
+	};
+});
+// "failAfter"
+ASQ.extend("failAfter",function $$extend(api,internals){
+	return function $$failAfter(num) {
+		var args = arguments.length > 1 ?
+			ARRAY_SLICE.call(arguments,1) :
+			void 0
+		;
+		num = +num || 0;
+
+		api.then(function $$then(done){
+			setTimeout(function $$set$timeout(){
+				done.fail.apply(ø,args);
+			},num);
+		});
+
+		return api;
+	};
+});
+
+ASQ.failAfter = function $$fail$after() {
+	return ASQ().failAfter.apply(ø,arguments);
+};
+// "first"
+ASQ.extend("first",function $$extend(api,internals){
+	return function $$first() {
+		if (internals("seq_error") || internals("seq_aborted") ||
+			arguments.length === 0
+		) {
+			return api;
+		}
+
+		var fns = ARRAY_SLICE.call(arguments);
+
+		api.then(function $$then(done){
+			function reset() {
+				error_messages.length = 0;
+			}
+
+			function success(trigger,idx,args) {
+				if (!finished) {
+					finished = true;
+
+					// first successful segment triggers
+					// main sequence to proceed as success
+					trigger(
+						args.length > 1 ?
+						ASQ.messages.apply(ø,args) :
+						args[0]
+					);
+
+					reset();
+				}
+			}
+
+			function failure(trigger,idx,args) {
+				if (!finished &&
+					!(idx in error_messages)
+				) {
+					completed++;
+					error_messages[idx] =
+						args.length > 1 ?
+						ASQ.messages.apply(ø,args) :
+						args[0]
+					;
+
+					// all segments complete without success?
+					if (completed === fns.length) {
+						finished = true;
+
+						// send errors into main sequence
+						error_messages.length = fns.length;
+						trigger.fail.apply(ø,error_messages);
+
+						reset();
+					}
+				}
+			}
+
+			var completed = 0, error_messages = [], finished = false,
+				sq = ASQ.apply(ø,ARRAY_SLICE.call(arguments,1))
+			;
+
+			wrapGate(sq,fns,success,failure,reset);
+
+			sq.pipe(done);
+		});
+
+		return api;
+	};
+});
+// "go-style CSP"
+"use strict";
```

Comment:
__[jshint]__
* Expected an assignment or function call and instead saw an expression.

<!-- Comment by [SideCI](https://sideci.com) -->